File: examples/java/alias1/Slave.java
/*
 * $Id: Slave.java 5059 2007-11-19 20:01:59Z mquinson $
 *
 * Copyright 2006,2007 Martin Quinson, Malek Cherier         
 * All rights reserved. 
 *
 * This program is free software; you can redistribute it and/or modify it
 * under the terms of the license (GNU LGPL) which comes with this package. 
 */

import simgrid.msg.*;

public class Slave extends simgrid.msg.Process 
{
	public void main(String[] args) throws JniException, NativeException 
	{
		Msg.info("Hello !");
		
		while(true) 
		{ 
			Task t = Task.receive(Host.currentHost().getName());	
		
			if (t instanceof FinalizeTask) 
			{
				break;
			}
			
			BasicTask task = (BasicTask)t;
			Msg.info("Received \"" + task.getName() + "\" ");
		
			Msg.info("Processing \"" + task.getName() +  "\" ");	 
			task.execute();
			Msg.info("\"" + task.getName() + "\" done ");
		}
		
		Msg.info("Received Finalize. I'm done. See you!");
	}
}

File: examples/cxx/basic/Slave.cxx
#include "Slave.hpp"
#include "FinalizeTask.hpp"
#include <Host.hpp>
#include <HostNotFoundException.hpp>

#include <iostream>
using namespace std;

#include <Msg.hpp>

MSG_IMPLEMENT_DYNAMIC(Slave, Process)

int Slave::main(int argc, char** argv)
{
	info("Hello");;
	
	while(true) 
	{ 
		Task* t = Task::get(0);	
	
		if(t->isInstanceOf("FinalizeTask")) 
		{
			delete t;
			break;
		}
		
		info(TEXT_("Received \"") + TEXT_(t->getName()) + TEXT_("\" "));
	
		info(TEXT_("Processing \"") + TEXT_(t->getName()));

		t->execute();
		
		info(TEXT_("\"") + TEXT_(t->getName()) + TEXT_("\" done "));

		delete t;
	}
	
	info("Received Finalize. I'm done. See you!");
	
	delete this;

	return 0;
}
